src/main/java/com/synopsys/integration/alert/component/authentication/actions/AuthenticationTestAction.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.component.authentication.actions;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.opensaml.saml2.metadata.provider.MetadataProviderException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.saml.metadata.ExtendedMetadataDelegate;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.action.TestAction;
+import com.synopsys.integration.alert.common.exception.AlertFieldException;
+import com.synopsys.integration.alert.common.message.model.MessageResult;
+import com.synopsys.integration.alert.common.persistence.accessor.FieldAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationFieldModel;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.component.authentication.descriptor.AuthenticationDescriptor;
+import com.synopsys.integration.alert.component.authentication.descriptor.AuthenticationUIConfig;
+import com.synopsys.integration.alert.web.security.authentication.ldap.LdapManager;
+import com.synopsys.integration.alert.web.security.authentication.saml.SAMLManager;
+import com.synopsys.integration.exception.IntegrationException;
+
+@Component
+public class AuthenticationTestAction extends TestAction {
+    private Logger logger = LoggerFactory.getLogger(AuthenticationTestAction.class);
+    private LdapManager ldapManager;
+    private SAMLManager samlManager;
+
+    @Autowired
+    public AuthenticationTestAction(LdapManager ldapManager, SAMLManager samlManager) {
+        this.ldapManager = ldapManager;
+        this.samlManager = samlManager;
+    }
+
+    @Override
+    public MessageResult testConfig(String configId, FieldModel fieldModel, FieldAccessor registeredFieldValues) throws IntegrationException {
+        logger.info("Testing authentication.");
+        boolean ldapEnabled = registeredFieldValues.getBooleanOrFalse(AuthenticationDescriptor.KEY_LDAP_ENABLED);
+        boolean samlEnabled = registeredFieldValues.getBooleanOrFalse(AuthenticationDescriptor.KEY_SAML_ENABLED);
+        if (ldapEnabled) {
+            performLdapTest(fieldModel, registeredFieldValues);
+        }
+
+        if (samlEnabled) {
```

Comment:
Will we want to say something if both are enabled? Or is that allowed?

Reply:
It's really only one or the other not both.  That's a validation check that happens.